src/server/project.ts

```diff
@@ -142,6 +154,35 @@ namespace ts.server {
             return this.cachedUnresolvedImportsPerFile;
         }
 
+        public static resolveModule(moduleName: string, initialDir: string, host: ServerHost, log: (message: string) => void): {} {

```

Comment:
Why not just reuse the module resolver the TS compiler uses [elsewhere](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/moduleNameResolver.ts#L296)? I understand that the module resolver is a lot more complicated thanks to its exposure the the compiler options, but it _should_ be configurable to load just JS, right? At the very least, it'd probably be better not to duplicate the folder traversal/lookup logic in two places?